server/jobs/export_grades.py

```diff
@@ -65,6 +68,48 @@ def export_student_grades(student, assignments):
     return student_row
 
 
+def collect_all_scores(assignments, user_ids):
+    all_scores = {}
+
+    for assign in assignments:
+        scores = (
+            db.session.query(Score.user_id, Score.kind, func.max(Score.score))
+            .filter(
+                Score.user_id.in_(user_ids),
+                Score.assignment_id == assign.id,
+                Score.archived == False,
+            )
+            .group_by(Score.user_id, Score.kind)
+            .order_by(Score.score)
+            .all()
+        )
+
+        members = GroupMember.query.filter(
```

Comment:
any reason to use db.session.query in the first case but GroupMember.query in the second case?

Reply:
GroupMember.query is used when you want to get the member objects back, db.session.query is used when you want arbitrary columns. I need to use it here since I'm using an aggregation function to get the max.